src/extension.ts

```diff
@@ -69,23 +69,36 @@ function didOpenTextDocument(
   if (!folder) {
     return;
   }
+
   if (
+    workspace
+      .getConfiguration()
+      .get<boolean>('rust-client.enableMultiProjectSetup', false)
+  ) {
+    folder = getCargoTomlWorkspace(folder, document.uri.fsPath);
+  } else if (
     workspace
       .getConfiguration()
       .get<boolean>('rust-client.nestedMultiRootConfigInOutermost', true)
   ) {
     folder = getOuterMostWorkspaceFolder(folder);
   }
-  // folder = getCargoTomlWorkspace(folder, document.uri.fsPath);
+
   if (!folder) {
     stopSpinner(`RLS: Cargo.toml missing`);
     return;
   }
 
-  if (!workspaces.has(folder.uri)) {
+  const folderPath = folder.uri.toString();
+
+  if (!workspaces.has(folderPath)) {
     const workspace = new ClientWorkspace(folder);
-    workspaces.set(folder.uri, workspace);
+    activeWorkspace = workspace;
+    workspaces.set(folderPath, workspace);
     workspace.start(context);
+  } else {
+    const ws = workspaces.get(folderPath);
+    activeWorkspace = typeof ws === 'undefined' ? null : ws;
```

Comment:
```typescript
const workspace = workspaces.get(folderPath);
activeWorkspace = (typeof workspace === 'undefined')
    ? null 
    : workspace;
```

Reply:
the formatter, which is build server checked, decides stuff like that.